lib/si-data-nats/src/jetstream/context.rs

```diff
@@ -0,0 +1,109 @@
+use async_nats::jetstream::stream::{Config, DeleteStatus, Stream};
+use async_nats::{jetstream, HeaderMap};
+use bytes::Bytes;
+use telemetry::prelude::*;
+
+use crate::jetstream::{Consumer, JetstreamResult};
+use crate::Client;
+use crate::HeaderName;
+
+/// Used in a [`HeaderMap`] for consumers of stream to reply back to requesters (via a reply mailbox and outside of the
+/// stream).
+pub static REPLY_SUBJECT_HEADER_NAME: HeaderName = HeaderName::from_static("X-Reply-Subject");
```

Comment:
When consumers of a stream need to reply, they will likely do so outside of a Jetstream and do it via regular NATS with a generated reply mailbox (which NATS knows how to garbage collect).